server.py

```diff
@@ -79,35 +84,69 @@ def sensor_log(cmd, data):
     global logging_sensor_data
     global logged_sensors
     global run_start_time
+    global output_files
+    global current_run_name
+    global run_names
 
     if cmd == 'start':
+        run_finished()
         pod_runs += 1
-        f = open('static/data/run_' + str(pod_runs) + ".csv",'w')
-        f.write('sensor,time,value\n')
-        f.close()
         run_start_time = datetime.now()
+        run_name = 'run_' + str(pod_runs) + "_" + run_start_time.strftime("%H-%M-%S")
+        current_run_name = run_name
+        f = open('static/logs/' + run_name + ".csv",'w')
+        f.write('sensor,time,value\n')
+        output_files[run_name] = f
+        for s in logged_sensors:
+            output_files[s] = open(output_files[s].name, 'a')
         logging_sensor_data = True
-        socketio.emit('log_update', {'pod_runs':pod_runs,'sensors':logged_sensors})
 
     elif cmd == 'update' and logging_sensor_data:
         time_crt = datetime.strftime(datetime.now(), '%H:%M:%S')
         time_passed = datetime.strftime(server_start_time + (datetime.now() - run_start_time), '%H:%M:%S')
 
-        f = open('static/data/run_' + str(pod_runs) + ".csv", 'a')
         data['rpm_scaled'] = data['rpm'] / 100
 
         for s in logged_sensors:
-            f.write(s + ',' + str(time_crt) + ',' + str(data[s]) + '\n')
-        f.close()
+            output_files[current_run_name].write(s + ',' + str(time_crt) + ',' + str(data[s]) + '\n')
+            output_files[s].write(current_run_name + ',' + str(time_passed) + ',' + str(data[s]) + '\n')
 
+        if data['velocity'] <= 0:
+            close_log_files()
+
+
+def run_finished():
+    global logging_sensor_data
+    global logged_sensors
+    global output_files
+    global current_run_name
+    global run_names
+    global deleted_runs
+
+    logging_sensor_data = False
+    try:
+        output_files[current_run_name].close()
         for s in logged_sensors:
-            f = open('static/data/' + str(s) + ".csv",'a')
-            f.write('run_' + str(pod_runs) + ',' + str(time_passed) + ',' + str(data[s]) + '\n')
-            f.close()
+            output_files[s].close()
+    except:
+        pass
 
+    if len(deleted_runs) > 0:
+        for s in logged_sensors:
+            f = open('static/logs/' + str(s) + ".csv",'r')
+            lines = f.readlines()
+            f.close()
```

Comment:
1. The creation of the 'logs' directory will be fixed in the next commit.

2. I don't think that could work, actually, because I want to read from the file, close it, then write back to it. If I use the "with" statement, Python handles the closing for me, so I either need to change the scope of 'lines' or I will be reading and writing from the same file at the same time, which I believe work but probably shouldn't do.